src/mbgl/util/tile_cover.cpp

```diff
@@ -127,8 +127,8 @@ std::vector<UnwrappedTileID> tileCover(const Point<double>& tl,
 
 } // namespace
 
-int32_t coveringZoomLevel(double zoom, SourceType type, uint16_t size) {
-    zoom += std::log(util::tileSize / size) / std::log(2);
+uint32_t coveringZoomLevel(double zoom, SourceType type, uint16_t size) {
+    zoom = std::max(0., zoom) + std::log(util::tileSize / size) / std::log(2);
```

Comment:
Should this `max` call include the addition (which accounts for satellite tile size differences)? Otherwise, it could be impossible to use z0 for satellite tiles.